flocker/control/httpapi.py

```diff
@@ -664,6 +664,89 @@ def saved(_):
         saving.addCallback(saved)
         return saving
 
+    @app.route("/configuration/containers/<name>", methods=['POST'])
+    @user_documentation(
+        """
+        Update a named container's configuration.
+
+        This will lead to the container being relocated to the specified host
+        and restarted. This will also update the primary host of any attached
+        datasets.
+        """,
+        examples=[u"move container"],
+    )
+    @structured(
+        inputSchema={
+            '$ref':
+            '/v1/endpoints.json#/definitions/configuration_container_update',
+        },
+        outputSchema={
+            '$ref':
+            '/v1/endpoints.json#/definitions/configuration_container',
+        },
+        schema_store=SCHEMAS,
+    )
+    def update_containers_configuration(self, name, host):
+        """
+        Update the specified container's configuration.
+
+        :param unicode name: A unique identifier for the container within
+            the Flocker cluster.
+
+        :param unicode host: The address of the node on which the container
+            will run.
+
+        :return: An ``EndpointResponse`` describing the container which has
+            been updated.
+        """
+        deployment = self.persistence_service.get()
+
+        for node in deployment.nodes:
+            for application in node.applications:
+                if application.name == name:
+                    target_node = self._find_node_by_host(host, deployment)
+                    # We only need to perform a move if the node currently
+                    # hosting the container is not the node it's already on.
```

Comment:
Oops, this comment doesn't make any sense. Obvious what I meant to write but I'll quickly do a commit to correct it.